Python/context.c

```diff
@@ -113,6 +113,10 @@ context_event_name(PyContextEvent event) {
 static void
 notify_context_watchers(PyThreadState *ts, PyContextEvent event, PyObject *ctx)
 {
+    // The callbacks are registered on the interpreter, not on the thread, so
+    // the only way callbacks can know which thread changed is by calling the
+    // callbacks from the affected thread.
+    assert(ts == _PyThreadState_GET());
```

Comment:
Could we assert that it's non-NULL as well? Someone could possibly skip over this check by not holding the GIL.
```suggestion
    assert(ts != NULL);
    assert(ts == _PyThreadState_GET());
```